components/renku_data_services/platform/api.spec.yaml

```
@@ -67,6 +161,45 @@ components:
         incident_banner:
           $ref: "#/components/schemas/IncidentBanner"
       additionalProperties: false
+    UrlRedirectPlanList:
+      description: A list of redirect info objects
+      type: array
+      items:
+        $ref: "#/components/schemas/UrlRedirectPlan"
+    UrlRedirectPlan:
+      description: A mapping from src URLs to targets, either Renku entities or URLs
+      type: object
+      properties:
+        etag:
+          $ref: "#/components/schemas/ETag"
+        source_url:
+          description: The protocol-relative URL to redirect
+          type: string
+        target_url:
+          description: The protocol-relative URL to redirect to
+          type: string
+      required:
+        - etag
+        - source_url
+        - target_url
+    UrlRedirectPlansGetQuery:
+      description: Query params for redirect list get request
+      allOf:
+      - $ref: "#/components/schemas/PaginationRequest"
+    UrlRedirectPlanPost:
+      description: Schema for creating a new redirect info
+      type: object
+      properties:
+        source_url:
+          description: The protocol-relative URL to redirect
+          type: string
+        target_url:
+          description: The protocol-relative URL to redirect to
+          type: string
+      required:
+        - source_url
+        - target_url
+      additionalProperties: false
```

A few things:
1. I am not sure what "protocol-relative" means exactly. Like the protocol is not needed in the target url?
2. Providing examples on the Post and on the get response is a good idea imo.


```suggestion
    UrlRedirectPlanPost:
      description: Schema for creating a new redirect info
      type: object
      properties:
        source_url:
          description: The protocol-relative URL to redirect
          type: string
        target_url:
          description: The protocol-relative URL to redirect to
          type: string
      required:
        - source_url
        - target_url
      additionalProperties: false
      example:
        source_url: "https://gitlab.renkulab.io/user/project"
        target_url: "https://github.com/user/project"
```

---

Fixed in 8448063860f8b36a0bc2423e161a2a2f78ef45f7